web/modules/custom/hoeringsportal_audit_log/src/Form/SettingsForm.php

```diff
@@ -0,0 +1,197 @@
+   *   The route provider service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   * @param \Drupal\hoeringsportal_audit_log\Helpers\ConfigHelper $configHelper
+   *   The configuration helper.
+   */
+  public function __construct(ConfigFactoryInterface $configFactory, protected RouteProviderInterface $routeProvider, protected EntityTypeManagerInterface $entityTypeManager, protected ConfigHelper $configHelper) {
+    parent::__construct($configFactory);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'hoeringsportal_audit_log_config';
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @phpstan-return array<mixed, mixed>
+   */
+  protected function getEditableConfigNames(): array {
+    return [
+      self::SETTINGS,
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @phpstan-param array<mixed, mixed> $form
+   * @phpstan-return array<mixed, mixed>
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    // Form constructor.
+    $form = parent::buildForm($form, $form_state);
+    // Default settings.
+    $config = $this->config(self::SETTINGS);
+
+    $enabledAuditIds = $this->configHelper->getEnabledAuditIds();
+
+    $form['info'] = [
+      '#markup' => $this->t('This configuration handles <em>when</em> to create logs, <a href=":os2web_audit_settings_url">edit the <code>os2web_audit</code> configuration</a>.', [
+        ':os2web_audit_settings_url' => Url::fromRoute('os2web_audit.plugin_settings_local_tasks')->toString(TRUE)->getGeneratedUrl(),
+      ]),
+    ];
+
+    $form['logged_pages'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Logged pages'),
+      '#tree' => TRUE,
+    ];
+
+    $form['logged_pages']['logged_route_names'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('Route names'),
+      '#default_value' => $config->get('logged_route_names') ? $this->fromArrayToString($config->get('logged_route_names')) : NULL,
+      '#description' => $this->t('Route names (one per line) to log when users visit, they can look like this: <code>hoeringsportal_citizen_proposal.admin_supporter</code> or <code>node.add</code>, if in doubt, ask your friendly neighborhood programmer.'),
+    ];
+
+    /** @var \Drupal\Core\Entity\EntityInterface[] $definitions */
+    $definitions = $this->entityTypeManager->getDefinitions();
+
+    $form['types'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Definition routes'),
+      '#tree' => TRUE,
+    ];
+
+    foreach ($definitions as $definitionId => $definition) {
+      if (in_array($definitionId, $enabledAuditIds)) {
+
+        $bundleEntityType = $definition->getBundleEntityType();
+
+        if ($bundleEntityType) {
+          $storage = $this->entityTypeManager->getStorage($bundleEntityType);
+          $types = $storage->loadMultiple();
+        }
+        else {
+          $types = [$definition];
+        }
```

Comment:
It may (sooner or later) come in handy with a comment telling what is going on here, and in general it may be worth telling what the structure of the configuration is.